### DISCUSSION TITLE
Renesas RA4M1 port?

### DISCUSSION THREAD

**Discussion by phil-barrett** [grblHAL/core#330] [General]
While I think the Arduino Uno R4 is a very late to the party product, it is possible that it could be a decent platform for grblHAL.  The negatives (to me anyway) are that it has an Arduino Uno form factor and, honestly, Renesas.  On the plus side, the RA4M1 Cortex M4 is a reasonably capable chip and there may be clones of it that sell for $10-15 USD.  Also, the RA4M1 supports 5V logic so it may help reduce interface costs for stepper drivers and spindle control. And, maybe there will be a nano form factor version

I have to say the 5x5 LED thing is kind of dumb.

My issue with Renesas is that they really are not attuned to anything other than industrial customers.  On several of the microcontroller forums I frequent there is widespread skepticism that they will sell into the hobbyist market.  Things like sparkfun, adafruit and other clones won't happen. No "pills" or Feathers or other derivative products. It is the same reason that true Raspberry Pi cones haven't happened (Broadcom is similar to Renesas).

I'd like to hear what others think about this.

By the way, in one of the Facebook CNC groups, there have been several people say that now there is a 32 bit Arduino, Grbl is available on it.  I gently point out that a) this is not the first 32 bit Arduino and b) 8-bit Grbl will take some porting - it's not a recompile by any stretch.  I also point out that there are several 32 bit Arduinos that grblHAL actually runs on. There even appears to be a CNC product based on the Due.

**Comment by terjeio** [comment]
> I have to say the 5x5 LED thing is kind of dumb.

Trying to compete with the [BBC micro:bit](https://microbit.org/)?

> I'd like to hear what others think about this.

Why I do not want to write a grblHAL driver:
* Arduino IDE - I'll admit I have not tried v2 which claims to support debugging. And v2 has no source control integration that I am aware of.
* The Renesas IDE (Eclipse based, a plus for me) is a paid for product. PlatformIO (from VSCode) _can_ be used instead but I find it (VSCode) clumsy.
* WiFi is over [a UART channel](https://github.com/platformio/platformio-core/issues/4028#issuecomment-1622317146) (?). This will severely limit any network functionality/making it harder to implement as the [Networking plugin](https://github.com/grblHAL/Plugin_networking) cannot be used.
* The grblHAL supported STM32 Nucleo-64 and Nucleo-144 boards has UNO connectivity. No 5V outputs but has 5V tolerant inputs.
* I do not want to spend time learning yet another MCU architecture.

However, I am willing to support anyone who decide to write a driver.

**Comment by andrewmarles** [comment]
I have the PicoHAL hardware very close to release that is a bit more optimized for CNC applications:

https://github.com/Expatria-Technologies/Plugins_picoHAL

While I am personally focused on using it as an extension board for GRBLHAL and LinuxCNC systems, it is fully capable of running GRBLHAL with CNC or Makershield.

![image](https://github.com/grblHAL/core/assets/6061539/72256091-531b-4ba0-9b07-673a83cd866b)
